raiden/network/proxies/payment_channel.py

```diff
@@ -205,28 +193,22 @@ def all_events_filter(
         # These topics must not be joined with the channel_filter, otherwise
         # the filter ChannelSettled wont match (observed with geth
         # 1.8.11-stable-dea1ce05)
-        unlock_topic1 = construct_event_topic_set(
-            event_unlock_abi,
-            [
-                self.participant1,
-                self.participant2,
-                locksroot,
-                unlocked_amount,
-                returned_amount,
-            ],
-        )
-        unlock_topic2 = construct_event_topic_set(
-            event_unlock_abi,
-            [
-                self.participant2,
-                self.participant1,
-                locksroot,
-                unlocked_amount,
-                returned_amount,
-            ],
+
+        event_unlock_abi = CONTRACT_MANAGER.get_event_abi(
+            CONTRACT_TOKEN_NETWORK,
+            EVENT_CHANNEL_UNLOCKED,
         )
-        unlock_topics = unlock_topic1
-        unlock_topics.extend(unlock_topic2)
+
+        event_unlock_topic = encode_hex(event_abi_to_log_topic(event_unlock_abi))
+        participant1_topic = encode_hex(self.participant1.rjust(32, b'\0'))
+        participant2_topic = encode_hex(self.participant2.rjust(32, b'\0'))
```

Comment:
This `rjust` is done because `encode_single`, when encoding an `address` to `bytes32` pads it with zeroes to the right, while events topics encoding is padded with zeroes to the left. Anyone knows of a standard utils function to do that?

Reply:
Also, `construct_event_topic_set` were returning a list with a single sublist containing the ordered arguments of the event to be ordered, which pratically just matched the first topic (event's log topic, the hash of the signature of the event) against all parameters, which were useless, and specifically, if included a `None` (our case), matched everything. We may look into checking that function, and opening an issue or PR on web3' repo to fix it.